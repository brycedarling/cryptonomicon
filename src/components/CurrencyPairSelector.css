button {
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  border-radius: 5px;
  background-image: -webkit-gradient(linear, left bottom, left top, color-stop(0.16, rgb(207, 207, 207)), color-stop(0.79, rgb(252, 252, 252)));
  background-image: -moz-linear-gradient(center bottom, rgb(207, 207, 207) 16%, rgb(252, 252, 252) 79%);
  background-image: linear-gradient(to top, rgb(207, 207, 207) 16%, rgb(252, 252, 252) 79%);
  padding: 3px;
  margin-right: 10px;
  border: 1px solid #000;
  color: black;
  text-decoration: none;
  outline: none;
}

.selected {
  border-color: #0000ff;
  background-image: -webkit-gradient(linear, left bottom, left top, color-stop(0.16, steelblue), color-stop(0.79, rgb(252, 252, 252)));
  background-image: -moz-linear-gradient(center bottom, steelblue 16%, rgb(252, 252, 252) 79%);
  background-image: linear-gradient(to top, steelblue 16%, rgb(252, 252, 252) 79%);
}
