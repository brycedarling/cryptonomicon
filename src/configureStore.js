import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const configureStore = () => {
  const initialState = {};

  const enhancers = [];

  const middleware = [
    thunk,
  ];

  if (process.env.NODE_ENV === 'development') {
    const { __REDUX_DEVTOOLS_EXTENSION__ } = window;

    if (typeof devToolsExtension === 'function') {
      enhancers.push(__REDUX_DEVTOOLS_EXTENSION__());
    }
  }

  const composedEnhancers = compose(
    applyMiddleware(...middleware),
    ...enhancers,
  );

  const store = createStore(
    rootReducer,
    initialState,
    composedEnhancers,
  );

  return store;
};

export default configureStore;
