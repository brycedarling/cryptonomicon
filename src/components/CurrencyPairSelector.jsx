import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { actions } from '../reducers';
import './CurrencyPairSelector.css';

const CurrencyPairSelector = props => (
  <div>
    <button
      className={props.currencyPair === 'USDT_BTC' ? 'selected' : ''}
      onClick={() => props.setCurrencyPair('USDT_BTC')}
    >
      BTC
    </button>
    <button
      className={props.currencyPair === 'USDT_ETH' ? 'selected' : ''}
      onClick={() => props.setCurrencyPair('USDT_ETH')}
    >
      ETH
    </button>
    <button
      className={props.currencyPair === 'USDT_LTC' ? 'selected' : ''}
      onClick={() => props.setCurrencyPair('USDT_LTC')}
    >
      LTC
    </button>
    <span>
      {(props.loading ? 'Loading...' : '')}
    </span>
  </div>
);

CurrencyPairSelector.propTypes = {
  loading: PropTypes.bool.isRequired,
  currencyPair: PropTypes.string.isRequired,
  setCurrencyPair: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  loading: state.ticks.ui.loading,
  currencyPair: state.ticks.ui.currencyPair,
});

const mapDispatchToProps = {
  setCurrencyPair: actions.setCurrencyPair,
};

export default connect(mapStateToProps, mapDispatchToProps)(CurrencyPairSelector);
