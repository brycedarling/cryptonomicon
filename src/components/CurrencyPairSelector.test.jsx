import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CurrencyPairSelector from './CurrencyPairSelector';
import configureStore from '../configureStore';

describe('components', () => {
  describe('CurrencyPairSelector', () => {
    it('renders correctly', () => {
      const store = configureStore();

      const tree = renderer.create((
        <Provider store={store}>
          <CurrencyPairSelector />
        </Provider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
