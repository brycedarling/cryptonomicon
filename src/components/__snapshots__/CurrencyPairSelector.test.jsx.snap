// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components CurrencyPairSelector renders correctly 1`] = `
<div>
  <button
    className="selected"
    onClick={[Function]}
  >
    BTC
  </button>
  <button
    className=""
    onClick={[Function]}
  >
    ETH
  </button>
  <button
    className=""
    onClick={[Function]}
  >
    LTC
  </button>
  <span>
    
  </span>
</div>
`;
